include/internal/catch_approx.cpp

```diff
@@ -55,18 +55,18 @@ namespace Detail {
         return marginComparison(m_value, other, m_margin) || marginComparison(m_value, other, m_epsilon * (m_scale + std::fabs(m_value)));
     }
 
-    void Approx::setMargin(double margin) {
-        CATCH_ENFORCE(margin >= 0,
-            "Invalid Approx::margin: " << margin << '.'
+    void Approx::setMargin(double margin_) {
```

Comment:
What is this supposed to be shadowed with?

Reply:
It's actually warning against the margin member function:
```
/tmp/catch2/include/internal/catch_approx.cpp:58:41: error: declaration of ‘margin’ shadows a member of ‘Catch::Detail::Approx’ [-Werror=shadow]
     void Approx::setMargin(double margin) {
                                         ^
In file included from /tmp/catch2/include/internal/catch_approx.cpp:9:0:
/tmp/catch2/include/internal/catch_approx.h:98:17: note: shadowed declaration is here
         Approx& margin( T const& newMargin ) {
                 ^~~~~~
```

I tested with gcc-8.2.0 (I've upgraded my system compiler since January) and this warning isn't being emitted.

Reply:
Okay, thanks.